sound/soc/sof/intel/hda.c

```diff
@@ -1045,12 +1045,32 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 			 */
 			unique_id = SDW_UNIQUE_ID(adr);
 			if (link->num_adr == 1 ||
-			    ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID ||
 			    ids[j].id.unique_id == unique_id) {
 				dev_dbg(bus->dev,
 					"found %x at link %d\n",
 					part_id, link_id);
 				break;
+			} else if (ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID) {
+				unsigned int part_id2;
+
```

Comment:
if (ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID) => no two or more of this codec. please check sdw_acpi_find_slaves. So why do you need to do the following check ?

Reply:
> if (ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID) => no two or more of this codec. please check sdw_acpi_find_slaves. So why do you need to do the following check ?

@RanderWang `if (ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID)` means there is no other codecs on **ACPI table**. But, we can have more than one codecs on `snd_soc_acpi_mach`'s `links`.
For example,
There are 2 rt1308 on [rt1308_1_dual_adr[]](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/common/soc-acpi-intel-tgl-match.c#L56). [link_slaves_found](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.c#L1016) will return `true` due to the [ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.c#L1044) condition when there is only one rt1308 on the ACPI table.